python/_tskitmodule.c

```diff
@@ -9646,6 +9646,80 @@ TreeSequence_k_way_stat_method(TreeSequence *self, PyObject *args, PyObject *kwd
     return ret;
 }
 
+static PyObject *
+TreeSequence_weighted_stat_vector_method(
+    TreeSequence *self, PyObject *args, PyObject *kwds, weighted_vector_method *method)
+{
+    PyObject *ret = NULL;
+    static char *kwlist[]
+        = { "weights", "windows", "mode", "span_normalise", "centre", NULL };
+    PyObject *weights = NULL;
+    PyObject *windows = NULL;
+    PyArrayObject *weights_array = NULL;
+    PyArrayObject *windows_array = NULL;
+    PyArrayObject *result_array = NULL;
+    tsk_size_t num_windows;
+    npy_intp *w_shape, result_shape[3];
+    tsk_flags_t options = 0;
+    tsk_size_t num_samples;
+    char *mode = NULL;
+    int span_normalise = true;
+    int centre = true;
+    int err;
+
+    if (TreeSequence_check_state(self) != 0) {
+        goto out;
+    }
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "OO|sii", kwlist, &weights, &windows,
+            &mode, &span_normalise, &centre)) {
+        goto out;
+    }
+    if (parse_stats_mode(mode, &options) != 0) {
```

Comment:
stats_mode and windows likewise need to provoke the error path in test_lowlevel.py

Reply:
They do - that error path is a bit different, though: https://github.com/tskit-dev/tskit/pull/2980/files#diff-dd8e543618311de68e41669a0abf93fd978875005d4c97f1f10d9dc199befb22R2712 -- `TSK_ERR_UNSUPPORTED_STAT_MODE` is raised in `c/tskit/trees.c` in general, not here.